TKeazirian.HTTPServer/Handler/SimpleOptionsHandler2.cs

```diff
@@ -0,0 +1,31 @@
+namespace TKeazirian.HTTPServer.Handler;
+
+using Response;
+using Request;
+
+public class SimpleOptionsHandler2 : Handler
+{
+    public override List<string> AllowedHttpMethods()
+    {
+        return new List<string>() { "GET", "PUT", "POST" };
+    }
+
+    public override Response HandleResponse(Request request)
+    {
+        if (request.GetRequestMethod() == "PUT" || request.GetRequestMethod() == "POST")
+        {
+            return new ResponseBuilder()
+                .SetStatusCode(HttpStatusCode.NotImplemented)
```

Comment:
What would happen if there was a PATCH or DELETE request to this handler? Do you think you could generalize the `NotImplemented` behavior for all unimplemented request methods?

Reply:
Well, what happens now is that it won't even enter that Handler if the request method isn't included in the AllowedMethods for that Handler.  So, if it is included (like PUT/POST here), you'll get a 501 and if it isn't included (like a PATCH or DELETE), it will give you a 404.  Are you saying you would prefer that behavior to change, in general?

Reply:
As in, should I be adding a handler for 501 Not Implemented and have that be the Handler called when the method isn't included in the AllowedHttpMethods, and then the 404 Not Found method is called in....other cases?  When the path doesn't exist, perhaps?

If so, would that be out of scope of this PR and added as a separate ticket?

Reply:
To me this looks like it's introducing a bug into the code, where the client could get an incorrect/misleading error status (404, should be 501). So I think you need to handle it here, at least in some way.

How is 501 similar in nature to 404? How are you handling 404's generically in a way that you're not currently handling 501's? Could you handle 501's more like that?

Remember when we talked about how eventually, not having your HTTP methods be config would start to cause difficulties? I think we might be at that point now.

Reply:
Yes,  I can definitely see that!  Can I punt it to complete this as is and then work on separate refactoring ticket to implement that?

Reply:
I also refactored it a bit more so I think it's clearer...

501 --> if the RoutesConfig contains the path, but the Method isn't Allowed (resource exists but that method isn't implemented)
404 --> if the RoutesConfig does NOT contain the path at all (resource doesn't exist)

But I agree that will definitely need a further refactor to introduce the Method/HTTP Verb along with the path and Handler...